Bluetooth: Rework the HCI header set

carlescufi opened the pull request:
Bluetooth: Rework the HCI header set

    In order to have clean, self-contained HCI headers that do not have any
    dependencies towards the Host or any other part of the system (except
    types), reorganize the headers in the following way:

    - Split out the macros and structs from hci.h to a new hci_types.h
    - Merge the existing hci_err.h into the new hci_types.h

Fixes #58214.

jhedberg commented on subsys/bluetooth/controller/hci/hci.c:
```diff
@@ -21,6 +21,7 @@
 #include <zephyr/bluetooth/hci_vs.h>
 #include <zephyr/bluetooth/buf.h>
 #include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
```
Why should the controller depend on conn.h? Isn’t that a purely host-specific API? What we discussed last week was that hci.h should get split into two files, so that one contains strictly only HCI protocol definitions (structs & defines) and nothing else.

carlescufi replied:
Will rework as per the Discord discussion.

carlescufi committed:
Bluetooth: Rework the HCI header set

In order to have clean, self-contained HCI headers that do not have any
dependencies towards the Host or any other part of the system (except
types), reorganize the headers in the following way:

- Split out the macros and structs from hci.h to a new hci_types.h
- Merge the existing hci_err.h into the new hci_types.h

Fixes #58214.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Rework internal header includes

Rework internal header files to not have includes, rather
have the required includes in the c source files.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak commented:
@carlescufi  I did some more clean up in Controller in a new commit in this PR: 7af43d235ff66cd0bde8a7d7deb7a93994865e8f

cvinayak commented:
Controller does continue to include `<zephyr/bluetooth/conn.h`  and `<zephyr/bluetooth/iso.h` (in c source files, not in include headers) for connection role definitions and ISO packing (sequential or interleaved), respectively. These exceptions in the Controller C source file should be addressed outside the scope of this PR. Otherwise, the changes in this PR looks good to me.

kruithofa reviewed (approved):
LGTM
